src/corefx/System.Globalization.Native/calendarData.cpp

```diff
@@ -6,11 +6,20 @@
 #include <string.h>
 #include <vector>
 
-#include "config.h"
+#include "icushim.h"
 #include "locale.hpp"
 #include "holders.h"
 #include "errors.h"
 
+#ifndef FEATURE_FIXED_ICU_VERSION
+// The System.Globalization.Native.so is built in mode when it can use any supported version
+// of ICU installed on the target machine and not just the one it was built with. So we can 
+// only use stable APIs and the UDAT_STANDALONE_SHORTER_WEEKDAYS doesn't belong to those.
+// We also want to support ICU versions where the UDAT_STANDALONE_SHORTER_WEEKDAYS was not 
+// defined yet.
+#undef HAVE_UDAT_STANDALONE_SHORTER_WEEKDAYS
```

Comment:
I thought I have tried to build it with ICU 52 and the U_HIDE_DRAFT_API defined and the UDAT_STANDALONE_SHORTER_WEEKDAYS was not present, so I have assumed it is still not stable in 52. Now I have tried again and it builds fine, so I think I was just daydreaming.
But with this new idea of using the U_ICU_VERSION_MAJOR_NUM as the minimum major version, it seems HAVE_UDAT_STANDALONE_SHORTER_WEEKDAYS should be handled the same way as the other one as you say.
Regarding
>If we try to take version of System.Globalization.Native.so built on ICU 51 and run it on a platform with ICU 50 or earlier, our loop to find a library to dlopen would fail, because we stop searching at U_ICU_VERSION_MAJOR_NUM right?

Yes, that would be the case.